src/tests/unit/stream_element_tests.js

```diff
@@ -196,3 +196,33 @@ test("test action=refresh discarded when matching request id", async () => {
 
   assert.ok(document.body.hasAttribute("data-modified"))
 })
+
+test("action=morph", async () => {
+  const templateElement = createTemplateElement(`<h1 id="hello">Hello Turbo Morphed</h1>`)
+  const element = createStreamElement("morph", "hello", templateElement)
+
+  assert.equal(subject.find("div#hello")?.textContent, "Hello Turbo")
+
+  subject.append(element)
+  await nextAnimationFrame()
+
+  assert.notOk(subject.find("div#hello"))
+  assert.equal(subject.find("h1#hello")?.textContent, "Hello Turbo Morphed")
+})
+
+test("action=morph with data-turbo-morph-style='innerHTML'", async () => {
+  const templateElement = createTemplateElement(`<h1 id="hello-child-element">Hello Turbo Morphed</h1>`)
+  const element = createStreamElement("morph", "hello", templateElement)
+  const target = subject.find("div#hello")
+  assert.equal(target?.textContent, "Hello Turbo")
+  target.setAttribute("data-turbo-morph-style", "innerHTML")
+
+  subject.append(element)
+
+  await nextAnimationFrame()
+
+  assert.ok(subject.find("div#hello"))
+  assert.ok(subject.find("div#hello > h1#hello-child-element"))
+  assert.equal(subject.find("div#hello > h1#hello-child-element").textContent, "Hello Turbo Morphed")
+})
```

Comment:
We'd also need tests covering how Stimulus listeners can handle events. See tests in `page_refresh_tests.js`.

Reply:
Hey @jorgemanrubia, could you expand a little on what exactly we are looking to test, targeting a controller element with a morph stream action? or just a controller listening to the morph events?

Reply:
Hey @jorgemanrubia, just checking in to see if you've had a chance to look at my previous question. It's the final piece I need to wrap up this PR. Thanks a lot!

Reply:
Sorry about the delay here @omarluq. What we need to test are two things:

- That the stream action triggers the events we expect. You can see an example of how we do that in the general case here:

https://github.com/hotwired/turbo/blob/e8d25b29460d745955dc9c0f11f046eac54996dc/src/tests/functional/page_refresh_tests.js#L38-L45

- That the `before-` events that the stream action emits give the listener a chance to cancel the action. Example here:

https://github.com/hotwired/turbo/blob/e8d25b29460d745955dc9c0f11f046eac54996dc/src/tests/functional/page_refresh_tests.js#L47-L60

The PR is looking great. Thanks for your work here.

Reply:
@jorgemanrubia No worries, thank you for your guidance, I pushed new browser test to cover both. Please lmk if you have anymore feedback.